### PR TITLE
Comments for PIOc_write_darray()

### PR DESCRIPTION
**Pull request by edhartnett**
I have added comments to PIOc_write_darray(). Please review the comments and tell me what I got wrong. ;-)

I have used double-question marks where I am confused. If you could provide some comments for those places, that would really speed things up.

This is related to #89.

### CODE REVIEW

File: src/clib/pio_darray.c [hunk 1]
```diff
@@ -361,34 +382,38 @@ int PIOc_write_darray(const int ncid, const int vid, const int ioid,
...
     if (arraylen > 0)
-        if (!(wmb->data = bgetr(wmb->data, (1+wmb->validvars)*arraylen*tsize)))
+        if (!(wmb->data = bgetr(wmb->data, (1 + wmb->validvars) * arraylen * tsize)))
             piomemerror(*ios, (1 + wmb->validvars) * arraylen * tsize, __FILE__, __LINE__);
 
     /* Get memory for ??? */
     if (!(wmb->vid = (int *)bgetr(wmb->vid, sizeof(int) * (1 + wmb->validvars))))
         piomemerror(*ios, (1 + wmb->validvars) * sizeof(int), __FILE__, __LINE__);
 
+    /* Get memory for ??? */
```
Comment (jedwards4b) [inline src/clib/pio_darray.c]: wmb->frame is the record number, we assume that the variables in the wmb list may not all have the same unlimited dimension value although they usually do.

File: src/clib/pio_darray.c [hunk 2]
```diff
@@ -407,32 +432,27 @@ int PIOc_write_darray(const int ncid, const int vid, const int ioid,
...
-      double asum=0.0;
-      printf("%s %d %d %d %d\n",__FILE__,__LINE__,vid,arraylen,iodesc->ndof);
-      for (int k=0;k<arraylen;k++){
-      asum += ((double *) array)[k];
-      }
-      printf("%s %d %d %g\n",__FILE__,__LINE__,vid,asum);
-      }
-    */
 
+    /* ??? */
```
Comment (jedwards4b) [inline src/clib/pio_darray.c]: add the unlimited dimension value of this variable to the frame array in wmb

File: tests/unit/test_darray.c [hunk 3]
```diff
@@ -1,7 +1,5 @@
-/**
- * @file
+/*
```
Comment (jedwards4b) [inline tests/unit/test_darray.c]: You've tried to simplify this test and that's fine, but instead of replacing a working test why not create a new one?

Comment (edhartnett) [inline tests/unit/test_darray.c]: This is not really a working test. If you look, you will see it does nothing with darrays (yet). It is just the start of the darray test I am writing. (The same is true of test_darray_async.c) They started as code copied from a test creating a more complex test file. In this first darray outing I am going to keep it very simple. ;-)

**Pull request merged**